docs/design/backups.md

```diff
@@ -0,0 +1,83 @@
+# Backup & Restore
+
+In particularly disastrous situations, we would like to be able to revert the etcd cluster to a previously known healthy state.
+This document lays out the design of how we want the healthy state to be captured, securely stored & safely restored to operator-managed etcd clusters.
+
+## Backup Resource
+
+We will introduce a new custom resource: `EtcdBackup`.
+
+```yaml
+apiVersion: etcd.improbable.io/v1alpha1
+kind: EtcdBackup
```

Comment:
How does this resource denote which cluster is being backed up?

Reply:
I would expect a `spec.clusterName` field.

Reply:
whoops missed this, thanks

Reply:
thoughts on using `clusterAddress` instead of `clusterName`? address might be more explicit as it would allow the backup to be taken from a different namespace, without needing another field to specify the cluster namespace

Reply:
I guess an address should really be similar to what you'd give to the `--endpoints` flag of `etcdctl`, which is actually a comma-separated list of member addresses. (e.g., `http://bees-0.bees.svc:2379,http://bees-1.bees.svc:2379,http://bees-2.bees.svc:2379`). Which we could do?